inst/qml/jaspPlotBuilder.qml

```diff
@@ -0,0 +1,2974 @@
+                        fieldWidth: 239
+                    }
+
+                    TextField {
+                        name: "dimensionText"
+                        id: dimensionText
+                        placeholderText: qsTr("Measurement dimension (e.g., cm, mg, minutes)")
+                        visible: yesRM.checked
+                        property bool active: yesRM.checked
+                        onActiveChanged: if (!active && count > 0) itemDoubleClicked(0)
+                        info: qsTr("Measurement dimension (e.g., cm, mg, minutes)")
+                        fieldWidth: 239
+                    }
+
+                    CheckBox {
+                        name:"useRMFactorAsFill"
+                        id:useRMFactorAsFill
+                        label: qsTr("Use RM factor as a group/color variable")
+                        visible: yesRM.checked
+                    }
+
+                    AssignedVariablesList {
+                        name: "variableYPlotBuilder"
+                        title: qsTr("Y axis")
+                        allowedColumns: ["scale", "ordinal", "nominal"]
+                        id: variableYPlotBuilder
+                        singleVariable: true
+                        visible: noRM.checked
+                        property bool active: noRM.checked
+                        onActiveChanged: if (!active && count > 0) itemDoubleClicked(0)
+                        info: qsTr("Select the variable for the Y-axis.")
+                    }
+
+                    AssignedVariablesList {
+                        name: "variableXPlotBuilder"
+                        title: qsTr("X axis")
+                        id: variableXPlotBuilder
+                        allowedColumns: ["scale", "ordinal", "nominal"]
+                        minLevels: 2
+                        singleVariable: true
+                        info: qsTr("Select the variable for the X-axis.")
+                        property bool active: noRM.checked
+                        onActiveChanged: if (!active && count > 0) itemDoubleClicked(0)
+                    }
+
+                    AssignedVariablesList {
+                        name: "variableColorPlotBuilder"
+                        title: qsTr("Group/Color by")
+                        id: variableColorPlotBuilder
+                        allowedColumns: ["scale", "ordinal", "nominal"]
+                        minLevels: 2
+                        singleVariable: true
+                        info: qsTr("Select the variable for data grouping, which will also determine the coloring..")
+                        onCountChanged: {
+                            if (count > 0) {
+                                colorByVariableX.checked = false;
+                                colorByVariableY.checked = false;
+                            }
+                        }
+                    }
+                    Group {
```

Comment:
Do I understand correctly that coloring can be done by either the variable assigned to the `Group/Color by` variable input or by the X or Y variable? Maybe changing the variable name to `Group Variable` and adding a radio button group below the input that changes the color mapping between `X Axis Variable` / `Y Axis Variable` / `Group Variable` with radio buttons enabled only if the corresponding variable is specified would make this clearer?